File: go/structural/Bridge/SuperSizeSoda.go
//SuperSizeSoda - two of two classes extending the Abstract


package bridge

import java.util.*
import org.apache.commons.lang3.stringUtils

public class SuperSizeSoda extends Soda {  
   public SuperSizeSoda() {
       setSodaImp()
   }
   
   public string pourSoda() {
       SodaImp sodaImp = this.getSodaImp()
		List retval = new ArrayList()
       for (int i = 0 i < 5; i++) {
           retval.add("...glug...")
           retval.add(sodaImp.pourSodaImp())
       }
       return stringUtils.join(retval, " ")
   }
}

File: scala/src/main/scala/Structural/Bridge/SuperSizeSoda.scala
//SuperSizeSoda - two of two classes extending the Abstract


package Structural.Bridge;

import java.util.*;
import org.apache.commons.lang3.StringUtils;

class SuperSizeSoda extends Soda {  
   SuperSizeSoda() {
       setSodaImp();
   }
   
   String pourSoda() {
       SodaImp sodaImp = this.getSodaImp();
		List retval = new ArrayList();
       for (int i = 0; i < 5; i++) {
           retval.add("...glug...");
           retval.add(sodaImp.pourSodaImp());
       }
       return StringUtils.join(retval, " ");
   }
}
